### PR TITLE
[RFR] Add Select field, translate options

### LINKED ISSUE #454: no default label translation for RadioButtonGroupInput
In RadioButtonGroupInput it seems that the default label translation is not implemented.

### PR DESCRIPTION
**Pull request by fzaninotto**
- [x] Add `<SelectField>`
- [x] Translate `<SelectField>` options by default (except in a Reference)
- [x] Translate `<SelectInput>` options by default (except in a Reference)
- [x] Translate `<AutocompleteInput>` options by default (except in a Reference)
- [x] Translate `<RadioButtonGroupInput>` options by default (except in a Reference)
- [x] Translate `<CheckboxGroupInput>` options by default (except in a Reference)
- [x] Add tests
- [x] Add documentation

Closes #454, #455.

### CODE REVIEW

**Comment by fzaninotto** [comment]
Switching to RFR

**Comment by jpetitcolas** [comment]
Needs rebase

File: src/mui/field/SelectField.js [hunk 1]
```diff
@@ -0,0 +1,118 @@
...
+ * ];
+ *
+ * However, in some cases (e.g. inside a `<ReferenceField>`), you may not want
+ * the choice to be translated. In that case, set the `translateChoice` prop to false.
+ * @example
+ * <SelectField source="gender" choices={choices} translateChoice={false}/>
+ *
+ * **Tip**: <ReferenceField> sets `translateChoice` to false by default.
+ */
+export const SelectField = ({ source, record = {}, choices, elStyle, optionValue, optionText, translate, translateChoice = true }) => {
```
Comment (jpetitcolas) [inline src/mui/field/SelectField.js]: Isn't the `translateChoice = true` redundant with the `defaultProps`? Same for `record`?

File: src/mui/input/CheckboxGroupInput.js [hunk 2]
```diff
@@ -60,38 +77,39 @@ export class CheckboxGroupInput extends Component {
         }
     };
 
-    render() {
+    renderCheckbox = (choice) => {
```
Comment (jpetitcolas) [inline src/mui/input/CheckboxGroupInput.js]: I would move this `renderXXX` method in another `<Checkbox />` component.

File: src/mui/input/RadioButtonGroupInput.js [hunk 3]
```diff
@@ -45,32 +47,58 @@ import Labeled from './Labeled';
...
+export class RadioButtonGroupInput extends Component {
     handleChange = (event, value) => {
         this.props.input.onChange(value);
     }
 
-    render() {
-        const { label, source, input, choices, optionText, optionValue, options, elStyle } = this.props;
-        const option = React.isValidElement(optionText) ? // eslint-disable-line no-nested-ternary
-            choice => React.cloneElement(optionText, { record: choice }) :
+    renderRadioButton = (choice) => {
```
Comment (jpetitcolas) [inline src/mui/input/RadioButtonGroupInput.js]: Same

File: src/mui/input/SelectInput.js [hunk 4]
```diff
@@ -47,20 +48,58 @@ import FieldTitle from '../../util/FieldTitle';
...
  */
-class SelectInput extends Component {
+export class SelectInput extends Component {
     handleChange = (event, index, value) => this.props.input.onChange(value);
 
-    render() {
-        const { allowEmpty, input, label, choices, optionText, optionValue, options, source, elStyle, meta: { touched, error }, resource } = this.props;
-        const option = React.isValidElement(optionText) ? // eslint-disable-line no-nested-ternary
-            choice => React.cloneElement(optionText, { record: choice }) :
+    renderMenuItem = (choice) => {
```
Comment (jpetitcolas) [inline src/mui/input/SelectInput.js]: Same

File: src/mui/input/CheckboxGroupInput.js (hunk 2 above)
Comment (fzaninotto) [inline src/mui/input/CheckboxGroupInput.js]: As it's not used outside of `<CheckboxGoupInput>`, and it's tested there, I think it's a hassle that we can avoid. Same for the two others.

**Comment by fzaninotto** [comment]
Review applied

**Pull request merged**